src/test/java/org/broadinstitute/consent/http/resources/ConsentVoteResourceTest.java

```diff
@@ -30,31 +32,125 @@
     private EmailNotifierService emailNotifierService;
 
     private ConsentVoteResource resource;
+    private Vote vote;
+    private Consent consent;
 
     @Before
     public void setUp() {
-        MockitoAnnotations.initMocks(this);
+        openMocks(this);
+
+        consent = new Consent();
+        consent.setConsentId(UUID.randomUUID().toString());
+
+        vote = new Vote();
+        vote.setVoteId(RandomUtils.nextInt(100, 1000));
+        vote.setVote(false);
+        vote.setRationale("Test");
     }
 
     private void initResource() {
         resource = new ConsentVoteResource(emailNotifierService, electionService, voteService);
     }
 
     @Test
-    public void testCreateConsentVote() throws Exception {
-        Consent consent = new Consent();
-        consent.setConsentId(UUID.randomUUID().toString());
-        Vote vote = new Vote();
-        vote.setVoteId(RandomUtils.nextInt(100, 1000));
-        vote.setVote(false);
-        vote.setRationale("Test");
+    public void testCreateConsentVoteSuccess() throws Exception {
         when(voteService.updateVoteById(any(), any())).thenReturn(vote);
         when(electionService.validateCollectEmailCondition(any())).thenReturn(true);
         doNothing().when(emailNotifierService).sendCollectMessage(any());
         initResource();
 
-        Response response = resource.firstVoteUpdate(vote, UUID.randomUUID().toString(), vote.getVoteId());
+        Response response = resource.firstVoteUpdate(vote, consent.getConsentId(), vote.getVoteId());
+        assertEquals(200, response.getStatus());
+    }
+
+    @Test
+    public void testCreateConsentVoteCollectMessageError() throws Exception {
+        when(voteService.updateVoteById(any(), any())).thenReturn(vote);
+        when(electionService.validateCollectEmailCondition(any())).thenReturn(true);
+        doThrow(new IOException()).when(emailNotifierService).sendCollectMessage(any());
+        initResource();
+
+        Response response = resource.firstVoteUpdate(vote, consent.getConsentId(), vote.getVoteId());
         assertEquals(200, response.getStatus());
+
+        // This test is passing, and it looks like it shouldn't. Does something need to be reworked in the resource class?
```

Comment:
This scenario looks like this should pass, which it does. There is a handler for catching IOException that intentionally does NOT trigger a downstream exception to the client. The catch logic goes all the way back to 10/21/2015 and is likely there to prevent unrelated email errors from being shown to the caller.